sycl/include/sycl/reduction.hpp

```diff
@@ -243,22 +243,23 @@ template <class Reducer> class combiner {
                : memory_scope::device;
   }
 
-  template <access::address_space Space, class T, class AtomicFunctor>
+  template <access::address_space Space, access::decorated IsDecorated, class T,
```

Comment:
Why adding `IsDecorated` ? you just need to pick one internally

this will also avoid using the legacy later on